### ISSUE TITLE
pop from empty list in certain YAML files

### ISSUE THREAD

**Issue by dcermak** [dcermak/salt-lsp#3]
I have encountered the following sls file, that causes the parser to fail with a `IndexError: pop from empty list`:
``` python
from salt_lsp.parser import parse

parse("""apache2:
   pkg.installed: []
   service.running:
     - enable: true
     - require:
       - pkg: apache2
""")
```
 which throws this exception:
```
IndexError                                Traceback (most recent call last)
<ipython-input-11-36bc9a975312> in <module>
----> 1 parse("""apache2:
      2   pkg.installed: []
      3   service.running:
      4     - enable: true
      5     - require:

~/suse/github.com/dcermak/salt_lsp/salt_lsp/parser.py in parse(document)
    687     :raises ValueException: for any other renderer but ``jinja|yaml``
    688     """
--> 689     return Parser(document).parse()

~/suse/github.com/dcermak/salt_lsp/salt_lsp/parser.py in parse(self)
    636             for token in tokens:
    637                 log.debug(token)
--> 638                 self._process_token(token)
    639         except yaml.scanner.ScannerError as err:
    640             log.debug(err)

~/suse/github.com/dcermak/salt_lsp/salt_lsp/parser.py in _process_token(self, token)
    525         if isinstance(token, yaml.BlockEndToken):
    526             last_start = self._block_starts.pop()
--> 527             last = self._breadcrumbs.pop()
    528             # pop breadcrumbs until we match the block starts
    529             closed = last

IndexError: pop from empty list
```

@cbosdo any ideas?

**Comment by dcermak** [comment]
I believe the culprit here is that lists created via `[]` are started via a `yaml.FlowSequenceStartToken` that is not tracked by the parser. I have a fix for that but it needs further testing.

**Comment by dcermak** [comment]
Fixed by #4.
